modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeFDSDataProviderCheck.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class JavaUpgradeFDSDataProviderCheck extends BaseJavaTermCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, JavaTerm javaTerm,
+			String fileContent)
+		throws Exception {
+
+		List<String> importNames = javaTerm.getImportNames();
+
+		String javaTermContent = javaTerm.getContent();
+
+		if (!importNames.contains(
+				"com.liferay.frontend.data.set.provider.FDSDataProvider")) {
+
+			return javaTermContent;
+		}
+
+		Matcher implementsFDSDataProviderMatcher =
+			_implementsFDSDataProviderPattern.matcher(javaTermContent);
+
+		if (implementsFDSDataProviderMatcher.find()) {
+			javaTermContent = _sortParametersMethods(javaTermContent);
+		}
+
+		Matcher parameterFDSDataProviderMatcher =
+			_parameterFDSDataProviderPattern.matcher(javaTermContent);
+
+		if (parameterFDSDataProviderMatcher.find()) {
+			javaTermContent = _sortParametersMethodCalls(
+				javaTermContent, parameterFDSDataProviderMatcher.group(1));
+		}
+
+		return javaTermContent;
+	}
+
+	@Override
+	protected String[] getCheckableJavaTermNames() {
+		return new String[] {JAVA_CLASS};
+	}
+
+	private boolean _checkClassName(
```

Comment:
@michaelmdelcavalcanti, please rename to `_hasClassName`, since it's a more common pattern in Liferay Source Code when the method returns a boolean. :wink:

Reply:
Fixed